### DISCUSSION TITLE
pac auth create not working with service principal

### DISCUSSION THREAD

**Discussion by alen1979** [microsoft/powerplatform-build-tools#698] [Q&A]
Hi
Im trying to use following command to create an auth profile using a service principal

`pac auth create --name MyOrg-SPN  --environment 00000000-0000-0000-0000-000000000000  --applicationId 00000000-0000-0000-0000-000000000000 --clientSecret 00000000-0000-0000-0000-000000000000 --tenant 00000000-0000-0000-0000-000000000000`

As result I'm getting this:

```

e0837553-c460-4c41-90c3-edb4905f0b27' authenticated successfully.
Validating connection...

Unable to validate connection to default environment.

Authentication profile created
    * UNIVERSAL MyOrg-SPN                                                               : e0837553-c460-4c41-90c3-edb4905f0b27     Public
```
I'm pretty sure the application user exists on my environment and is a System Administrator.

**Comment by rpothin** [comment]
Hi @alen1979,

I hope you are doing well.

If your service principal is not present as an application in your default environment, I will perhaps not work without specifying the "environment" input with the id of the environment where it has the System Administrator security role.

Could you try to run the command like below please?
```pwsh
pac auth create --name MyOrg-SPN --applicationId 00000000-0000-0000-0000-000000000000 --clientSecret $clientSecret --tenant 00000000-0000-0000-0000-000000000000 --environment < Your environment ID >
```

<img width="436" alt="image" src="https://github.com/microsoft/powerplatform-build-tools/assets/23240245/3c7a0de0-ffbe-4f27-b6cd-11b94e890966">

I hope it will unblock you.
Have a great day.
Regards,
Raph

**Reply by alen1979** [reply]
Hi @rpothin 

Thanks for your help.

I've been using the environment input parameter already.  See my initial post. 

Thanks
Alen

**Reply by rpothin** [reply]
Hi @alen1979,

Oh yes, I missed it.
I am sorry.

But I am curious, which pac command are you planning to use being connected with your service principal?

I am asking because I have done a quick search in my repositories and did not find a place with a "direct" connection with a service principal. I am mostly using the provided GitHub actions or Azure DevOps steps provided by Microsoft or different PowerShell tools depending on my needs.

Good luck in your search.
Have a great day.
Regards,
Raph
